.github/workflows/full_ci.yml

```diff
@@ -186,6 +186,34 @@ jobs:
       - name: Execute run-docker.sh
         run: sh ./ci/run-docker.sh ${{ matrix.target }}
 
+  build_channels_netbsd:
+    permissions:
+      contents: read # to fetch code (actions/checkout)
+
+    name: Build Channels NetBSD
+    runs-on: ubuntu-latest
+    env:
+      OS: netbsd
+    strategy:
+      fail-fast: true
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run netbsd vm
+        id: netbsd-ci
+        uses: vmactions/netbsd-vm@v1
+        with:
+          envs: 'OS'
+          usesh: true
+          prepare: |
+            /usr/sbin/pkg_add curl
+            curl https://sh.rustup.rs -sSf --output rustup.sh
+            sh rustup.sh -y --default-toolchain nightly --profile=minimal
+            . $HOME/.cargo/env
```

Comment:
It would probably be best to just adjust `ci/install-rust.sh` as needed (if at all) to do what is needed here.